[windows] Add Windows AppLocker Data Stream (MSI and Script)

nicpenning opened the pull request:
- Enhancement

## What does this PR do?

This PR adds the Windows AppLocker MSI and Script data stream which allows the ingestion of those events from the Windows Event Log. This also updates the dashboard with a better title and adds 4 new visualizations to explore the data.

Resolves Part of - https://github.com/elastic/integrations/issues/6979

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

![image](https://github.com/elastic/integrations/assets/5582679/4a520170-c6ee-49b3-bfe5-de662764b168)

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > The PR is not allowed to run in the CI yet
  


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-7279/detail/PR-7279/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-7279/detail/PR-7279/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-7279/detail/PR-7279/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-7279/detail/PR-7279/1//artifacts)   [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-08-05T09:42:31.896Z&rangeTo=2023-08-05T10:02:31.896Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=74ef1a0c586f6634b4425cd320b8bd10&transactionId=80bfbdc2435c2a31)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-08-05T09:52:31.896+0000


* Duration: 3 min 15 sec





</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![2](https://img.shields.io/badge/2%20-red)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-7279/detail/PR-7279/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `Load a resource file from a library`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-7279/runs/1/steps/81/log/?start=0">here</a></li>
<li>Description: <code>approval-list/elastic/integrations.yml</code></l1>
</ul>
  
  
  
  
##### `Error signal`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-7279/runs/1/steps/89/log/?start=0">here</a></li>
<li>Description: <code>githubApiCall: The REST API call https://api.github.com/orgs/elastic/members/nicpenning return the message : java.lang.Exception: httpRequest: Failure connecting to the service https://api.github.com/orgs/elastic/members/nicpenning : httpRequest: Failure connecting to the service https://api.github.com/orgs/elastic/members/nicpenning : Code: 404Error: {"message":"User does not exist or is not a member of the organization","documentation_url":"https://docs.github.com/rest/orgs/members#check-organization-membership-for-a-user"}</code></l1>
</ul>
  
  </p>
  </details>



## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

nicpenning commented:
The appropriate ingest pipelines were lost somewhere, I will fix.

elasticmachine commented:
Pinging @elastic/security-external-integrations (Team:Security-External Integrations)

nicpenning commented:
/test

andrewkroh commented:
/test

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`6/6`) | :green_heart: 
Files | 90.0% (`9/10`) | :-1: -10.0
Classes | 90.0% (`9/10`) | :-1: -10.0
Methods | 84.545% (`93/110`) | :+1: 11.818
Lines | 92.65% (`5458/5891`) | :-1: -7.35
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

efd6 commented on packages/windows/_dev/deploy/docker/files/config.yml:
```diff
@@ -219,3 +219,47 @@ rules:
+                  "sourcetype": "XmlWinEventLog:Security",
+                  "splunk_server": "69819b6ce1bd"
+              }
+          }
```
Final new line.

nicpenning replied:
So this should end with a new line?

nicpenning replied:
I think I fixed this with a commit

efd6 commented on packages/windows/data_stream/applocker_msi_and_script/agent/stream/httpjson.yml.hbs:
```diff
@@ -0,0 +1,104 @@
+{{/unless}}
+{{/unless}}
+response.decode_as: application/x-ndjson
+{{#if tags.length}}
+tags:
+{{else if preserve_original_event}}
+tags:
+{{/if}}
```
Are these conditions necessary?

nicpenning replied:
I am unsure. This was a copy from another Windows integration. I am unsure about what this does in the httpsjson.yml.hbs context.

efd6 replied:
If it's copied across, let's leave it as is.

nicpenning replied:
Okay, easy enough

efd6 commented on packages/windows/data_stream/applocker_msi_and_script/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,175 @@
+
+  - set:
+      field: ecs.version
+      value: '8.0.0'
```
Can we use a more up-to-date version number?

nicpenning replied:
I am all for that, yes.

nicpenning replied:
Are there checks that will occur if I bump this to 8.9.0?

efd6 replied:
It should just update the version in the documents.

nicpenning replied:
Okay, I will bump this up

nicpenning replied:
Done

nicpenning replied:
By the way, the docs were not updated when I ran the build. So my guess is that not much changed from 8.0.0 to 8.9.0 in this dataset.

efd6 reviewed (commented):
Can you provide screen shots showing the dashboard changes.

Also, please make the following change to CODEOWNERS
```diff
diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index ddcfae883..aacc76f5a 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -243,6 +243,7 @@
 /packages/websphere_application_server @elastic/obs-infraobs-integrations
 /packages/windows @elastic/elastic-agent-data-plane @elastic/security-external-integrations
 /packages/windows/data_stream/applocker_exe_and_dll @elastic/security-external-integrations
+/packages/windows/data_stream/applocker_msi_and_script @elastic/security-external-integrations
 /packages/windows/data_stream/forwarded @elastic/security-external-integrations
 /packages/windows/data_stream/perfmon @elastic/elastic-agent-data-plane
 /packages/windows/data_stream/powershell @elastic/security-external-integrations
```

efd6 commented:
/test

nicpenning commented:
> Can you provide screen shots showing the dashboard changes.
> 
> Also, please make the following change to CODEOWNERS
> 
> ```diff
> diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
> index ddcfae883..aacc76f5a 100644
> --- a/.github/CODEOWNERS
> +++ b/.github/CODEOWNERS
> @@ -243,6 +243,7 @@
>  /packages/websphere_application_server @elastic/obs-infraobs-integrations
>  /packages/windows @elastic/elastic-agent-data-plane @elastic/security-external-integrations
>  /packages/windows/data_stream/applocker_exe_and_dll @elastic/security-external-integrations
> +/packages/windows/data_stream/applocker_msi_and_script @elastic/security-external-integrations
>  /packages/windows/data_stream/forwarded @elastic/security-external-integrations
>  /packages/windows/data_stream/perfmon @elastic/elastic-agent-data-plane
>  /packages/windows/data_stream/powershell @elastic/security-external-integrations
> ```

Do you mean update the .png in the package or add those screenshots to this PR for review?

nicpenning commented:
I added these:
![image](https://github.com/elastic/integrations/assets/5582679/6970aca9-ff4f-48a2-b4ba-0939e5561467)

And removed the "-" in the title of the dashboard between [Windows Applocker] - Audited Blocked Applications
![image](https://github.com/elastic/integrations/assets/5582679/04a9051d-b6b1-42db-ae70-456808a9a876)

efd6 commented:
/test

efd6 reviewed (approved):
LGTM

elasticmachine commented:
Package windows - 1.30.0 containing this change is available at https://epr.elastic.co/search?package=windows
